extensions/wikia/AdEngine/js/slot/scrollHandler.js

```diff
@@ -0,0 +1,49 @@
+/*global define*/
+define('ext.wikia.adEngine.slot.scrollHandler', [
+    'ext.wikia.adEngine.adContext',
+    'wikia.log',
+    'wikia.document',
+    'wikia.window'
+], function (adContext, log, doc, win) {
+
+    var logGroup = 'ext.wikia.adEngine.slot.scrollHandler',
+        isRefreshed = {
+            PREFOOTER_LEFT_BOXAD: false,
+            PREFOOTER_RIGHT_BOXAD: false
+        };
+
+    function init() {
+        if (adContext.getContext().opts.enableScrollHandler) {
+            win.addEventListener('scroll', function () {
+                log('Scroll event listener has been added', 'debug', logGroup);
+                for (var slotName in isRefreshed) {
+                    refreshSlot(slotName);
```

Comment:
You're welcome :) It's a pain and in modern browsers you can just use `Object.keys()` to iterate through all object's properties. But it's not supported in IE8, though :(